src/open_inwoner/templates/export/profile/profile_export.html

```diff
@@ -36,22 +36,29 @@ <h1 class="h1">{{ object.get_full_name }}</h1>
         </tr>
         <tr>
             <th class="table__header">{% trans "Communicatievoorkeuren" %}</th>
-            <td class="table__item">{{ request.user.get_active_notifications }}</td>
+            <td class="table__item">{{ object.get_active_notifications }}</td>
         </tr>
     </table>
 
+    {% with pending_invitations=object.get_pending_invitations contacts_for_approval=object.get_contacts_for_approval active_contacts=object.get_active_contacts %}
+    {% if pending_invitations or contacts_for_approval or active_contacts %}
+        <h2 class="h2 h2--extra">{% trans "Netwerkcontacten" %}</h2>
+        {% include "export/include/contacts.html" %}
+    {% endif %}
+    {% endwith %}
 
-    <h2 class="h2 h2--extra">{% trans "Netwerkcontacten" %}</h2>
-    {% include "export/include/contacts.html" %}
-
-
-    <h2 class="h2 h2--extra">{% trans "Acties" %}</h2>
-    {% include "export/include/actions.html" with actions=object.actions.visible %}
-
+    {% with object.actions.visible as actions %}
+    {% if files %}
+        <h2 class="h2 h2--extra">{% trans "Acties" %}</h2>
+        {% include "export/include/actions.html" with actions=actions %}
+    {% endif %}
+    {% endwith %}
 
-    <h2 class="h2 h2--extra">{% trans "Bestanden" %}</h2>
-    {% if object.get_all_files %}
-        {% file_table files=object.get_all_files download_view="profile:documents_download" %}
```

Comment:
@pi-sigma Existing example of querying the database for a result-list but only using it for a boolean check, then discarding the results, and then calling the same exact function again (with all the queries en IO and everything) to actually process.